backend/app/services/active_user_service.rb

```diff
@@ -17,37 +17,46 @@ def self.active_user(request)
                     request
                 )
             utorid, _password = credentials.split(':')
-            user = User.find_by(utorid: utorid)
+            @user = User.find_by(utorid: utorid)
 
-            # Every user has the TA role
-            roles = user ? user.roles + %w[ta] : %w[ta]
-
-            roles += %w[instructor] if Instructor.find_by(utorid: utorid)
-            # Check if they are listed as an instructor for any course
-            if Rails.application.config.always_admin.include?(utorid)
-                roles += %w[admin]
-            end
-
-            return User.new(utorid: utorid, roles: roles)
+            return User.new(utorid: utorid, roles: grant_roles)
         end
 
         if (Rails.application.config.respond_to? :active_user_override) &&
                Rails.application.config.active_user_override
-            user = User.find_by(id: Rails.application.config.active_user_id)
-            return user if user
-
-            # If we're here, the database doesn't have the specified active user,
-            # so createa fake one with all the permissions. We *do not* save this
-            # user to the database.
-            return(
-                User.new(
+            @user = User.find_by(id: Rails.application.config.active_user_id)
+            if @user
+                return User.new(
+                  utorid: @user.utorid, roles: grant_roles
+              )
+            else
+                # If we're here, the database doesn't have the specified active user,
+                # so createa fake one with all the permissions. We *do not* save this
+                # user to the database.
+                return(
+                  User.new(
                     utorid: 'defaultactive', roles: %w[admin instructor ta]
                 )
-            )
+              )
+            end
         end
 
         # rubocop:disable Style/RaiseArgs
         raise NotImplementedError.new 'active_user Route is not implemented yet for production mode'
         # rubocop:enable Style/RaiseArgs
     end
+
+    def self.grant_roles
```

Comment:
I'm surprised Rubocop let this through, but I dont think you should have a `self` here. And maybe you should call it `granted_roles` or `computed_roles`

Reply:
Since the above `active_user` function was defined with `self` (`def self.active_user(request)`), therefore `granted_roles` needs to have `self` too. Otherwise it results in a NameError.